packages/gatsby/src/utils/webpack.config.js

```diff
@@ -423,6 +414,28 @@ module.exports = async (
       ],
     }
 
+    // if react-hot-reloading
+    if (
+      stage === `develop` &&
+      process.env.GATSBY_HOT_LOADER !== `fast-refresh`
+    ) {
+      resolve.alias[`react-hot-loader`] = path.dirname(
+        require.resolve(`react-hot-loader/package.json`)
+      )
+
+      // We need to add @hot-loader/react-dom to support hot-loader work with hooks
+      try {
+        resolve.alias[`react-dom/server$`] = require.resolve(`react-dom/server`)
+        resolve.alias[`react-dom`] = require.resolve(`@hot-loader/react-dom`)
+      } catch (err) {
+        const { version: reactDomVersion } = require(`react-dom/package.json`)
```

Comment:
`@hot-loader/react-dom` has versions starting from `16.7.0`. Technically we support `^16.4.2` so the warning is not actionable for users on react versions between those 2 - worth silencing the warning if version is below 16.7? There are no hooks there anyway (assuming this is only needed for hooks support)

Reply:
I'll do 16.8.0 because that's when hooks are a thing and you'll only need it for hooks.

Reply:
I don't know specific details but `React 16.6+ features may not work.` from the error/warning message implies it's not just hooks? (or maybe message is just not updated - not sure if I remember correctly but hooks were expected to land in 16.7 but didn't)

Reply:
Improved semver checks. These are all the patches that they have needed to do https://github.com/gaearon/react-hot-loader/blob/master/src/webpack/patch.js